packages/mapper/problem.go

```diff
@@ -101,3 +108,42 @@ func GetTagsList(problem model.Problem) []string {
 	}
 	return tagsList
 }
+
+func PutProblemPackage(slug string, pkgDir string) error {
+	ctx := context.Background()
+	minioClient := application.GetMinioClient()
+
+	// remove old package
+	objectsCh := minioClient.ListObjects(ctx, application.GetBucketName(), minio.ListObjectsOptions{
+		Prefix:    slug,
+		Recursive: true,
+	})
+	for objInfo := range objectsCh {
+		if objInfo.Err != nil {
+			return objInfo.Err
+		}
+
+		err := minioClient.RemoveObject(ctx, application.GetBucketName(), objInfo.Key, minio.RemoveObjectOptions{})
+		if err != nil {
+			return err
+		}
+	}
+
+	filepath.Walk(pkgDir, func(path string, info fs.FileInfo, err error) error {
+		if info.IsDir() {
+			return nil
+		}
+		relativePath := filepath.Join(slug, strings.Replace(path, pkgDir, "", 1))
+
+		_, minioErr := minioClient.FPutObject(ctx, application.GetBucketName(),
+			relativePath,
+			path,
+			minio.PutObjectOptions{})
+		if minioErr != nil {
+			log.Fatalln(minioErr)
+		}
+		return minioErr
+	})
+
+	return nil
+}
```

Comment:
@KanadeSiina 
Can be weried to put actions related with minio in `mapper`.

Put this part of code in `packages/agent/minio/problem.go` |  `packages/agent/minio.go`. (for choose which, depends on whether we use more than one bucket)
Which means the package is the broker of data transferring between server & minio.